src/pvecontrol/sanitycheck.py

```diff
@@ -0,0 +1,196 @@
+  INFO = 'INFO'
+  OK = 'OK'
+
+ICONS = {
+  CheckCode.CRIT.value: '❌',
+  CheckCode.WARN.value: '⚠️',
+  CheckCode.INFO.value: 'ℹ️',
+  CheckCode.OK.value: '✅',
+}
+
+class CheckMessage:
+  def __init__(self, code: CheckCode, message):
+    self.code = code
+    self.message = message
+
+  def display(self, padding_max_size):
+    padding = padding_max_size - len(self.message)
+    print(f"{self.message}{padding * '.'}{ICONS[self.code.value]}")
+
+  def __len__(self):
+    return len(self.message)
+
+class Check:
+
+  def __init__(self, type: CheckType, name: str, messages = None):
+    if messages is None:
+      messages = []
+    self.type = type
+    self.name = name
+    self.messages = messages
+
+  def add_messages(self, messages):
+    if isinstance(messages, CheckMessage):
+      self.messages.append(messages)
+    elif isinstance(messages, list):
+      self.messages += messages
+
+  def set_code(self, code: CheckCode):
+    self.code = code
+
+  def display(self, padding_max_size):
+    print(f"{fonts.BOLD}{self.name}{fonts.END}\n")
+    for msg in self.messages:
+      msg.display(padding_max_size)
+    print()
+
+class SanityCheck():
+
+    check_methods = (
+      "_validate_nodes_capacity",
+      "_validate_ha_groups",
+      "_validate_ha_vms",
+    )
+
+    check_display_order = [
+      CheckType.HA,
+      CheckType.Node,
+    ]
+
+    def __init__(self, proxmox: PVECluster):
+      self._proxmox = proxmox
+      self._ha = self._proxmox.ha()
+      self._checks = []
+
+    def run(self):
+      for method in SanityCheck.check_methods:
+        self._checks.append(getattr(self, method)())
+
+    def _get_longest_message(self):
+      size = 0
+      for check in self._checks:
+        for msg in check.messages:
+          if len(msg) > size:
+            size = len(msg)
+      return size + 1
+
+    def display(self):
+      size = self._get_longest_message()
+      current_type = None
+      for check in self._checks:
+        if current_type != check.type:
+          current_type = check.type
+          dash_size = int((size - (len(check.type.value) + 2))/2)
+          print(f"{dash_size*'-'} {check.type.value} {dash_size*'-'}\n")
+        check.display(size)
+
+    def _check_cpu_overcommit(self, maxcpu, cpufactor, allocated_cpu):
+      return (maxcpu *  cpufactor) <= allocated_cpu
+
+    def _check_mem_overcommit(self, max_mem, min_mem, allocated_mem):
+      return (allocated_mem +  min_mem) >= max_mem
+
+    # Check nodes capacity
+    def _validate_nodes_capacity(self):
+      node_config = self._proxmox.config['node']
+      check = Check(CheckType.Node, "Check Node capacity")
+      for node in self._proxmox.nodes:
+        if self._check_cpu_overcommit(node.maxcpu, node_config['cpufactor'], node.allocatedcpu):
+          msg = "Node %s is in cpu overcommit status: %s allocated but %s available"%(node.node, node.allocatedcpu, node.maxcpu)
+          check.add_messages(CheckMessage(CheckCode.CRIT, msg))
```

Comment:
Shouldn't this be a warning instead?

Reply:
https://github.com/enix/pvecontrol/pull/18#discussion_r1865913828